src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -171,6 +172,62 @@ public PackageBuilder GetPackageBuilder(IPackTaskRequest<IMSBuildItem> request)
                     request.RepositoryBranch,
                     request.RepositoryCommit);
             }
+
+            var hasLicenseExpression = !string.IsNullOrEmpty(request.PackageLicenseExpression);
+            var hasLicenseFile = !string.IsNullOrEmpty(request.PackageLicenseFile);
+            if (hasLicenseExpression || hasLicenseFile)
+            {
+                if (request.LicenseUrl != null)
```

Comment:
Not necesarilly as the code is written now since the PackTaskRequest does GetNullForEmpty, but it's more future proof and consistent so I will fix it.